app/assets/src/components/common/CoverageVizSidebar/CoverageVizSidebar.jsx

```diff
@@ -0,0 +1,367 @@
+
+  state = {
+    currentAccession: null
+  };
+
+  componentDidUpdate(prevProps, prevState) {
+    if (!prevProps.visible && this.props.visible) {
+      this.setCurrentAccession(this.props.params.accessionSummaries[2].id);
+    }
+
+    if (!prevState.currentAccessionData && this.state.currentAccessionData) {
+      this.renderHistogram(this.state.currentAccessionData);
+      this.renderGenomeViz(this.state.currentAccessionData);
+    }
+  }
+
+  getDataForAccession = async accessionId => {
+    if (this._accessionDataCache[accessionId]) {
+      return this._accessionDataCache[accessionId];
+    } else {
+      // Replace with network fetch.
+      const data = ACCESSION_COVERAGE_DATA[accessionId];
+
+      this._accessionDataCache[accessionId] = data;
+      return data;
+    }
+  };
+
+  loadAccession = async accession => {
+    const data = await this.getDataForAccession(accession.id);
+
+    this.setState({
+      currentAccessionData: data
+    });
+  };
+
+  setCurrentAccession = accessionId => {
+    const accession = find(
+      ["id", accessionId],
+      this.props.params.accessionSummaries
+    );
+
+    this.setState({
+      currentAccession: accession,
+      currentAccessionData: null
+    });
+    this.loadAccession(accession);
+  };
+
+  processCoverageData = (coverageData, coverageBinSize) =>
+    coverageData.map((value, index) => ({
+      x0: index * coverageBinSize,
+      length: value
+    }));
+
+  processContigsReadData = (contigs, reads) => {
+    return [
+      contigs.map(contig => alignRange([contig[1], contig[2]])),
+      reads.map(read => alignRange([read[1], read[2]]))
+    ];
+  };
+
+  renderHistogram = data => {
+    const coverageData = this.processCoverageData(
+      data.coverage,
+      data.coverage_bin_size
+    );
+
+    this.histogram = new Histogram(this.histogramContainer, [coverageData], {
+      labelY: "Coverage",
+      domain: [0, data.total_length],
+      skipBins: true,
+      showStatistics: false,
+      colors: ["#A9BDFC"],
+      barOpacity: 1,
+      margins: {
+        left: 170,
+        right: 40,
+        top: 30,
+        bottom: 30
+      },
+      numTicksY: 2,
+      labelYOffset: 15,
+      labelYLarge: true
+    });
+    this.histogram.update();
+  };
+
+  renderGenomeViz = data => {
+    const contigsReadsData = this.processContigsReadData(
+      data.contigs,
+      data.reads
+    );
+
+    this.contigsReadGenomeViz = new GenomeViz(
+      this.contigsReadsContainer,
+      contigsReadsData,
+      {
+        domain: [0, data.total_length]
+      }
+    );
+    this.contigsReadGenomeViz.update();
+  };
+
+  getAccessionOptions = () => {
+    return this.props.params.accessionSummaries.map(summary => ({
+      text: summary.id,
+      value: summary.id
+    }));
+  };
+
+  getAccessionMetrics = () => {
+    if (!this.state.currentAccessionData) {
+      return {};
+    }
+
+    const referenceAccession = (
+      <BasicPopup
+        trigger={
```

Comment:
we'll want to add analytics tracking here at some point... see https://github.com/chanzuckerberg/idseq-web/pull/2120 .

Reply:
Will do in follow-up PR.